### PR TITLE
Logger factory

### PR DESCRIPTION
**Pull request by ifesdjeen**
cc @smaldini 

I've implemented slf4j and java.utils.logging for now, which should be enough for the start. 

Let's merge that one and implement other ones if we need/want them. 
Also next logical step would be to make slf4j a compile time dependency instead of runtime.

```
BUILD SUCCESSFUL

Total time: 1 mins 59.461 secs
```

### CODE REVIEW

File: reactor-core/src/main/java/reactor/logging/LoggerFactory.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
+package reactor.logging;
+
+public abstract class LoggerFactory {
+
+    private static volatile LoggerFactory defaultFactory =
```
Comment (smaldini) [inline reactor-core/src/main/java/reactor/logging/LoggerFactory.java]: Why volatile here and not final ?

File: reactor-core/src/main/java/reactor/logging/Logger.java [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+ * WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package reactor.logging;
+
+/**
+ * Repackaged Logger for internal purposes. Will pick up the existing
+ * logger implementation. Refer Factory for more information.
+ */
+public interface Logger {
```
Comment (smaldini) [inline reactor-core/src/main/java/reactor/logging/Logger.java]: so far so good, I just think I will add an extra few utils on top like an extra argument to each logging level for specifying origin Object of the trace, will be useful for the pluggable monitor

**Comment by smaldini** [comment]
Dependency is already optional I think, there are some places where it is still compile on the outer rim I will track down with a mass local find. 
Overall its good slight suggestions to make it perfect:
- move to support
- use a maximum of inner class to hide away our logger factory internals in the source (private/package scope eventually for testing)

File: reactor-core/src/main/java/reactor/logging/LoggerFactory.java (hunk 1 above)
Comment (ifesdjeen) [inline reactor-core/src/main/java/reactor/logging/LoggerFactory.java]: True. Great catch!

File: reactor-core/src/main/java/reactor/logging/Logger.java (hunk 2 above)
Comment (ifesdjeen) [inline reactor-core/src/main/java/reactor/logging/Logger.java]: ok...

Comment (smaldini) [inline reactor-core/src/main/java/reactor/logging/Logger.java]: nat convinced ? I need it to correlate later on the monitoring stream ("track down who's component emitted this log"). It's purely optional and doesn't prevent loggers to be created statically, its just in addition of log.info(str) there would be an extra log.info(origin, str)

Comment (ifesdjeen) [inline reactor-core/src/main/java/reactor/logging/Logger.java]: DONE

Comment (ifesdjeen) [inline reactor-core/src/main/java/reactor/logging/Logger.java]: I'm mostly convinced about what you say.

Comment (ifesdjeen) [inline reactor-core/src/main/java/reactor/logging/Logger.java]: you're our thought leader and everytihng

**Comment by ifesdjeen** [comment]
@smaldini hope i've adressed your comments, re-squashed commits

**Pull request merged**